test/hack/resource/pkg/resourcetypes/eni.go

```diff
@@ -154,3 +160,59 @@ func (e *ENI) Cleanup(ctx context.Context, ids []string) ([]string, error) {
 
 	return deleted, errs
 }
+
+// get-vpc-resource-controller-enis is to get leaked ENIs by the vpc-resource-controller
+// Issue: https://github.com/aws/karpenter-provider-aws/issues/5582
+func (e *ENI) getVpcResourceControllerENIs(ctx context.Context, expirationTime time.Time, excludedClusters []string) (ids []string, err error) {
+	var nextToken *string
+	for {
+		out, err := e.cloudFormationClient.DescribeStacks(ctx, &cloudformation.DescribeStacksInput{
+			NextToken: nextToken,
+		})
+		if err != nil {
+			fmt.Println("dfdsf", err)
+			return ids, err
+		}
+
+		stacks := lo.Reject(out.Stacks, func(s cloudformationtypes.Stack, _ int) bool {
+			return s.StackStatus == cloudformationtypes.StackStatusDeleteComplete ||
```

Comment:
Why are we checking for the stack first here? Not immediately apparent. If it's needed, should we add a comment?